### PR TITLE
chore: Official Release & Repository Maintenance

### LINKED ISSUE #42: Is this repo maintained?
I noticed various bumped versions that need to be merged. I'm wondering if is there a plan to merge them?

### PR DESCRIPTION
**Pull request by arivera-xealth**
## Important Announcement
We are considering this an official release candidate since this has been working well internally for some time now. 

## Non-Breaking Changes

fixes #42

- adds `tsconfig.spec.json` so that eslint can parse the `test` directory appropriately
- addresses all and any relevant eslint errors
- minor type castings for typescript changes 4.9 and above
- upgrades package-lock.json to latest _node 18_ npm version
- removes `any` as an allowed type
- we are not upgrading to node18 because GHA is not compatible with node18 yet:
```
Error: System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter ''using: node18' is not supported, use 'docker', 'node12' or 'node16' instead.')
```

## Dependency upgrades:
bumps versions to latest for the following dev dependencies:
```json
    "@types/jest": "^29.5.1",
    "@types/lodash": "^4.14.194",
    "@types/node": "^18.16.1",
    "@typescript-eslint/parser": "^5.59.1",
    "@vercel/ncc": "^0.36.1",
    "eslint": "^8.39.0",
    "eslint-plugin-github": "^4.7.0",
    "eslint-plugin-jest": "^27.2.1",
    "jest": "^29.5.0",
    "jest-circus": "^29.5.0",
    "prettier": "^2.8.8",
    "ts-jest": "^29.1.0",
    "ts-node": "^10.9.1",
    "typescript": "^5.0.4"
```

bumps versions to latest for the following dependencies:
```json
    "@actions/core": "^1.10.0",
    "@actions/exec": "^1.1.1",
    "@actions/github": "^5.1.1",
    "@actions/io": "^1.1.3",
```

## Testing
- A full end-to-end test was conducted here: https://github.com/arivera-xealth/sample-repo/pull/85
- unit and integration tests are passing
- all files compile correctly and as expected 
- all types resolve correctly

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+module.exports = {
+  "plugins": ["jest", "@typescript-eslint"],
+  "extends": ["plugin:github/recommended", "plugin:@typescript-eslint/recommended-requiring-type-checking"],
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaVersion": 9,
+    "sourceType": "module",
+    "project": {
+      "project": true,
+      "tsconfigRootDir": __dirname
```
Comment (arivera-xealth) [inline .eslintrc.js:10]: Recommended by https://typescript-eslint.io/linting/typed-linting and https://typescript-eslint.io/linting/troubleshooting/#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file

File: src/github-helper.ts [hunk 2]
```diff
@@ -42,18 +42,18 @@ export async function createPullRequest(
     const title =
       github.context.payload &&
       github.context.payload.pull_request &&
-      github.context.payload.pull_request.title
-    core.info(`Using body '${title}'`)
+      (github.context.payload.pull_request.title as unknown as string)
+    core.info(`Using title '${title || ''}'`)
```
Comment (joebowbeerxealth) [inline src/github-helper.ts:46]:
```suggestion
    core.info(`Using title '${title ?? ''}'`)
```

Null coalescing operator is preferred here?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

File: src/github-helper.ts [hunk 3]
```diff
@@ -42,18 +42,18 @@ export async function createPullRequest(
...
 
     // Get PR body
     const body =
       github.context.payload &&
       github.context.payload.pull_request &&
-      github.context.payload.pull_request.body
-    core.info(`Using body '${body}'`)
+      (github.context.payload.pull_request.body as unknown as string)
+    core.info(`Using body '${body || ''}'`)
```
Comment (joebowbeerxealth) [inline src/github-helper.ts:53]:
```suggestion
    core.info(`Using body '${body ?? ''}'`)
```

File: src/github-helper.ts [hunk 4]
```diff
@@ -64,10 +64,10 @@ export async function createPullRequest(
 
     // Apply labels
     if (inputs.labels.length > 0) {
-      const prLabels: Label[] =
+      const prLabels =
```
Comment (joebowbeerxealth) [inline src/github-helper.ts:67]:
```suggestion
      const prLabels = github.context.payload?.pull_request?.labels ?? []
```

Null coalescence and optional chaining work well together.

File: src/github-helper.ts [hunk 5]
```diff
@@ -136,8 +141,11 @@ export function buildBranchesFromLabels(inputs: Inputs): string[] {
   const potentialBranches =
```
Comment (joebowbeerxealth) [inline src/github-helper.ts:141]: Simplify and tighten with null coalescence and optional chaining

**Review by joebowbeerxealth — COMMENTED** [review]
Looks good. A couple of suggestions.

**Comment by arivera-xealth** [comment]
@joebowbeerxealth updated with suggestions. e2e test also passed.

**Review by joebowbeerxealth — APPROVED** [review]
LGTM

**Pull request merged**